include/bio/var_io/reader_options.hpp

```diff
@@ -0,0 +1,206 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2021, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2021, Knut Reinert & MPI für molekulare Genetik
+// Copyright (c) 2020-2021, deCODE Genetics
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides bio::var_io::reader_options and various pre-defined field_types.
+ * \author Hannes Hauswedell <hannes.hauswedell AT decode.is>
+ */
+
+#pragma once
+
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/views/char_to.hpp>
+#include <seqan3/utility/type_list/traits.hpp>
+
+#include <bio/format/bcf.hpp>
+#include <bio/format/vcf.hpp>
+#include <bio/stream/transparent_istream.hpp>
+#include <bio/stream/transparent_ostream.hpp>
+#include <bio/var_io/dynamic_type.hpp>
+#include <bio/var_io/header.hpp>
+#include <bio/var_io/misc.hpp>
+
+/* TODO
+ *
+ * Simplify the field types and provide the following:
+ *
+ * field_types (default) with shallow/deep switch and all VCF except GENOTYPES
+ * field_types_vcf_style with no switch as shallow VCF including VCF-GENOTYPES
+ * field_types_vcf_style with no switch as shallow BCF including numbers
+ *
+ * add types for genotypes
+ */
+namespace bio::var_io
+{
+
+/*!\name Pre-defined field types
+ * \brief These can be used to configure the behaviour of the bio::var_io::reader via bio::var_io::reader_options.
+ * \{
+ */
+
+/*!\brief The default field types for variant io.
+ *!\ingroup var_io
+ *
+ * \details
+ *
+ * These traits define a record type with minimal memory allocations for all input formats.
+ * It is the recommended record type when iterating ("streaming") over files that ca be any variant IO format.
+ *
+ * The "style" of the record resembles the BCF specification, i.e. contigs, FILTERs and INFO identifiers are
+ * represented as numbers (not strings); and the genotypes are encoded by-genotype (not by-sample).
+ * See bio::var_io::genotypes_bcf_style for more information on the latter.
+ *
+ * \warning Shallow types
+ *
+ * These records are not self-contained, i.e. they depend on caches and will become invalid when the reader moves to
+ * the next record.
+ * Since some elements in the record are views, it may not be possible and/or safe to change all values.
+ */
+template <ownership own = ownership::shallow>
+inline constexpr auto field_types_bcf_style =
```

Comment:
I think it's unintuitive that default field types are in reader options but default field ids are in misc?

Reply:
default fields IDs are used by the writer as well, but the field types are only used by the reader 🤷🏻

Reply:
ugh, I see. well the user probably uses all.hpp headers anyway

Reply:
I can put them in misc anyway if you prefer.

Reply:
where is the wiki were we put design discussion? I'm not sure whats best yet as I just get into the code more deeply step by step.

Reply:
This is one of the things where "clean library design" suggests minimal includes and putting them with the reader_options but "element of least surprise" for the user would suggest putting them together with the IDs.

Reply:
Yepp. I don't know what weighs more in this specific case :shrug: